datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -381,7 +381,7 @@ impl LogicalPlanBuilder {
     }
 
     /// Apply an alias
-    pub fn alias(self, alias: impl Into<String>) -> Result<Self> {
+    pub fn alias(self, alias: impl Into<OwnedTableReference>) -> Result<Self> {
```

Comment:
👍  this change makes a lot of sense to me and is consistent with creating names of other relations (like `CreateExternalTable`)

https://docs.rs/datafusion-expr/22.0.0/datafusion_expr/logical_plan/struct.CreateExternalTable.html